geomesa-web/geomesa-web-data/src/main/scala/org/locationtech/geomesa/web/analytics/AnalyticEndpoint.scala

```diff
@@ -0,0 +1,331 @@
+import org.apache.commons.lang3.exception.ExceptionUtils
+import org.apache.spark.sql.types.StructType
+import org.geotools.data.DataStoreFinder
+import org.json4s.{DefaultFormats, Formats}
+import org.locationtech.geomesa.compute.spark.sql.GeoMesaSparkSql
+import org.locationtech.geomesa.utils.cache.FilePersistence
+import org.locationtech.geomesa.utils.classpath.ClassPathUtils
+import org.locationtech.geomesa.web.core.GeoMesaScalatraServlet
+import org.scalatra.json.NativeJsonSupport
+import org.scalatra.{BadRequest, InternalServerError, Ok}
+import org.slf4j.LoggerFactory
+
+import scala.collection.JavaConversions._
+import scala.io.Source
+import scala.xml.Elem
+
+/**
+ * Rest endpoint to access geomesa analytics
+ */
+class AnalyticEndpoint(persistence: FilePersistence) extends GeoMesaScalatraServlet with NativeJsonSupport {
+
+  import AnalyticEndpoint._
+  override protected implicit def jsonFormats: Formats = DefaultFormats
+
+  private val logger = LoggerFactory.getLogger(classOf[AnalyticEndpoint])
+  var debug = true
+
+  override val root: String = "analytics"
+
+  override def defaultFormat: Symbol = 'json
+
+  // TODO ensure hadoop settings are available on the classpath...
+
+  // register any persisted data stores with the sql engine
+  allDataStoreParams().values.foreach(GeoMesaSparkSql.registerDataStore)
+  sys.addShutdownHook(GeoMesaSparkSql.stop(0))
+
+  // jars that will be distributed with the spark job
+  lazy val distributedJars: Seq[String] = {
+    val url = Option(getClass.getClassLoader.getResource("spark-jars.list"))
+        .getOrElse(getClass.getClassLoader.getResource("spark-jars-default.list"))
+    val source = Source.fromURL(url)
+    val jars = try { source.getLines().toList } finally { source.close() }
+    val searchLocations = Iterator(
+      () => ClassPathUtils.getJarsFromClasspath(classOf[AnalyticEndpoint]),
+      () => ClassPathUtils.getJarsFromClasspath(classOf[HttpServletRequest]),
+      // jboss classloaders don't allow us to get jars out... instead just use the filesystem
+      () => ClassPathUtils.getJarsFromEnvironment("JBOSS_HOME")
+    )
+    ClassPathUtils.findJars(jars, searchLocations).map(_.getAbsolutePath)
+  }
+
+  // explicit reference to the spark jar - this is needed b/c spark can't find itself with jboss' classloading
+  lazy val jbossSparkJar = {
+    // this will match the default geomesa-assembly - for a custom install you can specify
+    // the jar along with the other configs, which will override this
+    val jbossJars = ClassPathUtils.getJarsFromEnvironment("JBOSS_HOME")
+    val sparkJar = jbossJars.find { jar =>
+      jar.getName.startsWith("spark-") && jar.getName.endsWith("-geomesa-assembly.jar")
+    }
+    sparkJar.map(j => Map("spark.yarn.jar" -> j.getAbsolutePath))
+  }
+
+  before() {
+    contentType = formats(format)
+  }
+
+  /**
+   * Registers a data store, making it available for spark sql queries
+   */
+  post("/ds/:alias") {
```

Comment:
Instead of doing this, should we have the bean take the Geoserver catalog as a constructor arg and process it for GeoMesa data stores?